openml-lightgbm/lightgbm-provider/src/test/java/com/feedzai/openml/provider/lightgbm/LightGBMBinaryClassificationModelTrainerTest.java

```diff
@@ -300,6 +301,153 @@ public void fitWithNonASCIIFeatureNameIsPossible () {
+        }
+
+        assertThat(average(num1.get(0)))
+                .as("num1 contribution average should be lower in negative class")
+                .isLessThan(average(num1.get(1)));
+
+        assertThat(average(cat1.get(0)))
+                .as("cat contribution average should be lower in negative class")
+                .isLessThan(average(cat1.get(1)));
+    }
+
+    /**
+     * Test Contributions.
+     *
+     * @throws URISyntaxException For errors when loading the dataset resource.
+     * @throws IOException        For errors when reading the dataset.
+     * @since 1.2.2
+     */
+    @Test
+    public void testFeatureContributionsTargetMiddle() throws URISyntaxException, IOException {
+        final Dataset dataset = CSVUtils.getDatasetWithSchema(
+                TestResources.getResourcePath(DATASET_RESOURCE_NAME),
+                TestSchemas.CATEGORICALS_SCHEMA_LABEL_AT_END,
+                10000
+        );
+        final int targetIndex = dataset.getSchema().getTargetIndex().get();
+        final int num1Index = 1;
+        final int cat1Index = 4;
+
+        final Map<String, String> trainParams = new HashMap<>(MODEL_PARAMS);
+        trainParams.replace(NUM_ITERATIONS_PARAMETER_NAME, "100");
+
+        final LightGBMBinaryClassificationModel model = new LightGBMModelCreator().fit(
+                dataset,
+                new Random(),
+                trainParams
+        );
+
+        final ArrayList<List<Double>> num1 = getListOfTwoLists();
+        final ArrayList<List<Double>> cat1 = getListOfTwoLists();
+
+        // fetch one negative classification
+        for (final Iterator<Instance> it_neg = dataset.getInstances(); it_neg.hasNext(); ) {
+            final Instance instance = it_neg.next();
+            final int classIndex = (int) instance.getValue(targetIndex);
+            final double[] featureContributions = model.getFeatureContributions(instance);
+
+            num1.get(classIndex).add(featureContributions[num1Index]);
+            cat1.get(classIndex).add(featureContributions[cat1Index]);
+        }
+
+        assertThat(average(num1.get(0)))
+                .as("num1 contribution average should be lower in negative class")
+                .isLessThan(average(num1.get(1)));
+
+        assertThat(average(cat1.get(0)))
+                .as("cat contribution average should be lower in negative class")
+                .isLessThan(average(cat1.get(1)));
+    }
+
+    /**
+     * Test Contributions.
+     *
+     * @throws URISyntaxException For errors when loading the dataset resource.
+     * @throws IOException        For errors when reading the dataset.
+     * @since 1.2.2
+     */
+    @Test
+    public void testFeatureContributionsTargetBeginning() throws URISyntaxException, IOException {
+        final Dataset dataset = CSVUtils.getDatasetWithSchema(
+                TestResources.getResourcePath(DATASET_RESOURCE_NAME),
+                TestSchemas.CATEGORICALS_SCHEMA_LABEL_AT_START,
```

Comment:
these three tests are basically the same except for this line. can we refactor this logic in a single method that receives as parameter `DatasetSchema`